app/nexus/internal/route/operator/status.go

```diff
@@ -0,0 +1,366 @@
+//
+// Source values:
+//   - "keeper": Root key available through keeper reconstruction
+//   - "unknown": Root key source cannot be determined
+func getRootKeyStatus() RootKeyStatus {
+	if isRootKeyAvailable() {
+		return RootKeyStatus{
+			Status: "AVAILABLE",
+			Source: "keeper",
+		}
+	}
+
+	return RootKeyStatus{
+		Status: "UNAVAILABLE",
+		Source: "unknown",
+	}
+}
+
+// getBackingStoreStatus tests the connectivity and performance of the
+// persistent storage backend. It measures response time and determines
+// connection health through basic operations.
+//
+// The function performs a lightweight health check and measures the
+// time taken to complete the operation, providing both connectivity
+// status and performance metrics.
+//
+// Status values:
+//   - "CONNECTED": Backing store is reachable and responsive
+//   - "DISCONNECTED": Backing store unavailable or unresponsive
+//
+// Response time is measured in milliseconds and included in the status
+// for performance monitoring and alerting purposes.
+func getBackingStoreStatus() BackingStore {
+	start := time.Now()
+
+	healthy := isBackingStoreHealthy()
+	responseTime := int(time.Since(start).Milliseconds())
+
+	status := "CONNECTED"
+	if !healthy {
+		status = "DISCONNECTED"
+	}
+
+	return BackingStore{
+		Status:         status,
+		Type:           getBackingStoreType(),
+		ResponseTimeMs: &responseTime,
+	}
+}
+
+// getBackingStoreType identifies the type of persistent storage backend
+// currently configured for the system. This information helps operations
+// teams understand the storage architecture and troubleshoot issues.
+//
+// The function queries the environment configuration to determine the
+// backend type and logs the detected type for debugging purposes.
+//
+// Common backend types include:
+//   - "postgres": PostgreSQL database backend
+//   - "mysql": MySQL database backend
+//   - "file": File system backend
+//   - "memory": In-memory backend (testing/development)
+func getBackingStoreType() string {
+	backendType := env.BackendStoreType()
+
+	log.Log().Debug("message", "Detected backing store type", "type", backendType)
+	return string(backendType)
+}
+
+// getSecretsCount returns the total number of secrets currently stored
+// in the backing store. This metric is useful for capacity planning,
+// monitoring system usage, and detecting unexpected changes in secret count.
+//
+// The function only returns a count when the backing store is healthy
+// to ensure the reported number is accurate and reliable. If the backing
+// store is unhealthy, nil is returned to indicate the count cannot be
+// determined.
+//
+// Returns:
+//   - *int: Pointer to secret count when backing store is healthy
+//   - nil: When backing store is unhealthy or inaccessible
+func getSecretsCount() *int {
+	secrets := state.ListKeys()
+	if isBackingStoreHealthy() {
+		count := len(secrets)
+		return &count
+	}
+	return nil
+}
```

Comment:
note that there is no backing store for "in memory" and "lite" modes.

Make sure this function handles that accordingly, or you parse the result properly at the receiving end of this function.